templates/ts/client-browser.ts

```diff
@@ -80,40 +80,48 @@ export class {{name}}Client {
       ...this.options,
       ...options,
     };
-    const response = await fetch(`${this.serverUrl}/{{name}}`, {
-      ...mergedOptions,
-      headers: {
-        ...mergedOptions.headers,
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify(body),
-      method: 'POST',
-    });
-    const isJSON = (response.headers.get('content-type') || '').startsWith('application/json');
+    let response: Response;
+    let responseBody: any;
+    let isJSON: boolean;
+    try {
+      response = await fetch(`${this.serverUrl}/{{name}}`, {
+        ...mergedOptions,
+        headers: {
+          ...mergedOptions.headers,
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(body),
+        method: 'POST',
+      });
+      isJSON = (response.headers.get('content-type') || '').startsWith('application/json');
+      if (isJSON && (response.status >= 200 && response.status < 300 || response.status === 400 || response.status === 500)) {
```

Comment:
Is it necessary to check the response status code?

Reply:
Added comment.

Reply:
I'm not sure we should limit parsing the response to these error codes, it makes this line too dependant on the other conditions and likely to break.